src/main.rs

```diff
@@ -181,6 +187,17 @@ fn run_profile(
     sample_rate: Option<u64>,
     native: bool,
 ) -> Result<(), ApplicationError> {
+    if cfg!(target_os = "macos") {
+        let effective_uid = get_effective_uid();
+        if effective_uid != 0 {
+            let args = env::args().collect::<Vec<_>>();
+            return Err(InsufficientPermissionsMacOSSnafu {
+                program_command: args.join(" "),
```

Comment:
You really want https://docs.rs/shlex/latest/shlex/fn.try_join.html here, I think. But I can also change that later :)